java/ql/src/semmle/code/java/frameworks/play/Play.qll

```diff
@@ -0,0 +1,155 @@
+/**
+ * Provides classes and predicates for working with the Play framework.
+ */
+
+import java
+
+/**
+ * The `play.mvc.Result` class.
+ */
+class PlayMvcResultClass extends Class {
+  PlayMvcResultClass() { this.hasQualifiedName("play.mvc", "Result") }
+}
+
+/**
+ * The `play.mvc.Results` class.
+ */
+class PlayMvcResultsClass extends Class {
+  PlayMvcResultsClass() { this.hasQualifiedName("play.mvc", "Results") }
+}
+
+/**
+ * The `play.mvc.Http$RequestHeader` class.
+ */
+class PlayMvcHttpRequestHeader extends RefType {
+  PlayMvcHttpRequestHeader() { this.hasQualifiedName("play.mvc", "Http$RequestHeader") }
+}
+
+/**
+ * A `play.mvc.BodyParser<>$Of` annotation.
+ */
+class PlayBodyParserAnnotation extends Annotation {
+  PlayBodyParserAnnotation() { this.getType().hasQualifiedName("play.mvc", "BodyParser<>$Of") }
+}
+
+/**
+ * A `play.filters.csrf.AddCSRFToken` annotation.
+ */
+class PlayAddCsrfTokenAnnotation extends Annotation {
+  PlayAddCsrfTokenAnnotation() {
+    this.getType().hasQualifiedName("play.filters.csrf", "AddCSRFToken")
+  }
+}
```

Comment:
These two annotation classes do not appear to be currently used. Is that intentional?

Reply:
This is intentional, this has use cases to find controllers invoking the annotation.